src/Core/src/Platform/iOS/ViewExtensions.cs

```diff
@@ -698,5 +698,115 @@ internal static void UpdateLayerBorder(this CoreAnimation.CALayer layer, IButton
 			if (stroke.CornerRadius >= 0)
 				layer.CornerRadius = stroke.CornerRadius;
 		}
+
+		internal static UIViewController? GetViewController<T>(this UIView view) where T : UIViewController
+		{
+			var nextResponder = view as UIResponder;
+			while (nextResponder is not null)
+			{
+				nextResponder = nextResponder.NextResponder;
+
+				if (nextResponder is T viewController)
+					return viewController;
+			}
+			return null;
+		}
+
+		internal static UIView? FindNextView(this UIView view, UIView superView, Type[] requestedTypes)
```

Comment:
Instead of a specific array of types here can we just do a `Func<UIView, bool>`  and let the caller define the conditional for a match? 

I think that would avoid the need for the type checks and simplify overall.